gobblin-temporal/src/main/java/org/apache/gobblin/temporal/ddm/work/WUProcessingSpecForBorrowingPriorState.java

```diff
@@ -0,0 +1,114 @@
+    super(fileSystemUri, workUnitsDir, eventSubmitterContext);
+  }
+
+  @Override
+  public boolean isToDoJobLevelTiming() {
+    return true;
+  }
+
+  @Override
+  public @NonNull EventSubmitterContext getEventSubmitterContext() {
+    // NOTE: We are using the metrics tags from Job Props to create the metric context for the timer and NOT
+    // the deserialized jobState from HDFS that is created by the real distcp job. This is because the AZ runtime
+    // settings we want are for the job launcher that launched this Yarn job.
+    try {
+      FileSystem fs = Help.loadFileSystemForce(this);
+      JobState jobState = Help.loadJobStateUncached(this, fs);
+      List<Tag<?>> tagsFromCurrentJob = this.getTags();
+      String metricsSuffix = this.getMetricsSuffix();
+      List<Tag<?>> tags = this.calcMergedTags(tagsFromCurrentJob, metricsSuffix, jobState);
+      return new EventSubmitterContext(tags, JobMetrics.NAMESPACE);
+    } catch (IOException ioe) {
+      throw new RuntimeException(ioe);
+    }
+  }
+
+  private List<Tag<?>> calcMergedTags(List<Tag<?>> tagsFromCurJob, String metricsSuffix, JobState jobStateFromHdfs) {
+    // Construct new tags list by combining subset of tags on HDFS job state and the rest of the fields from the current job
+    Map<String, Tag<?>> tagsMap = new HashMap<>();
+    Set<String> tagKeysFromJobState = new HashSet<>(Arrays.asList(
+        TimingEvent.FlowEventConstants.FLOW_NAME_FIELD,
+        TimingEvent.FlowEventConstants.FLOW_GROUP_FIELD,
+        TimingEvent.FlowEventConstants.FLOW_EXECUTION_ID_FIELD,
+        TimingEvent.FlowEventConstants.JOB_NAME_FIELD,
+        TimingEvent.FlowEventConstants.JOB_GROUP_FIELD));
+
+    // Step 1, Add tags from the AZ props using the original job (the one that launched this yarn app)
+    tagsFromCurJob.forEach(tag -> tagsMap.put(tag.getKey(), tag));
+
+    // Step 2. Add tags from the jobState (the original MR job on HDFS)
+    List<String> targetKeysToAddSuffix = Arrays.asList(TimingEvent.FlowEventConstants.FLOW_NAME_FIELD, TimingEvent.FlowEventConstants.FLOW_GROUP_FIELD);
+    GobblinMetrics.getCustomTagsFromState(jobStateFromHdfs).stream()
+        .filter(tag -> tagKeysFromJobState.contains(tag.getKey()))
+        .forEach(tag -> {
+          // Step 2a (optional): Add a suffix to the FLOW_NAME_FIELD AND FLOW_GROUP_FIELDS to prevent collisions when testing
+          String value = targetKeysToAddSuffix.contains(tag.getKey())
+              ? tag.getValue() + metricsSuffix
+              : String.valueOf(tag.getValue());
+          tagsMap.put(tag.getKey(), new Tag<>(tag.getKey(), value));
+        });
+
+    // Step 3: Overwrite any pre-existing metadata with name of the current caller
+    tagsMap.put(GobblinMetricsKeys.CLASS_META, new Tag<>(GobblinMetricsKeys.CLASS_META, getClass().getCanonicalName()));
+    return new ArrayList<>(tagsMap.values());
+  }
+}
```

Comment:
Does this make sense to move into `Help` or some other Util class rather than tying it into the WUProcessing spec? It seems generic enough and useful across classes.

Reply:
I didn't expect we'd ever use it long term, so I kept it here.  I moved these verbatim out of `ProcessWorkUnitWorkflowImpl` and am encapsulating this special case of needing to mess around w/ and ultimately "spoof" the tags entirely within this class, since I don't expect it to be needed going forward w/ our E2E flow.